### DISCUSSION TITLE
import csv files

### DISCUSSION THREAD

**Discussion by kkerklaan** [sqlitebrowser/sqlitebrowser#2740] [Q&A] [answered]
hello everyone! I am new to the whole programming world, but I want to import existing access file into sqlitebrowser. I know I can do this via export access, as a textfile and then change it to csv, but when I import it, every cell has added a '00;' how can I fix this? It also does not include the column name.
<img width="710" alt="access 1" src="https://user-images.githubusercontent.com/85932216/122028750-9f072480-cdcc-11eb-9980-5ff9b9d826e1.PNG">
<img width="593" alt="access 2" src="https://user-images.githubusercontent.com/85932216/122028755-9f9fbb00-cdcc-11eb-91d7-1d9007e35b91.PNG">

**Comment by chrisjlocke** [comment]
What have you exported it from Access as?  A CSV?  Any chance of opening that in Notepad and screenshotting it?  Was it via a report, or an export routine?
Also, in your first screenshot, you have a gemcode of a date, then under that is 'gdm96-2' - is this right?

**Reply by chrisjlocke** [reply]
I found an online tool that might assist.
https://www.rebasedata.com/convert-access-to-sqlite-online

**Reply by kkerklaan** [reply]
in notepad I see the same issue, so that means that it goes wrong with the export 
<img width="925" alt="access 3" src="https://user-images.githubusercontent.com/85932216/122032366-db884f80-cdcf-11eb-911e-8ab16e754440.PNG">

**Reply by kkerklaan** [reply]
yes, I exported it as a csv file, via the export button. I do not know if the date is right, these are not my databases, but from a company where I am doing an internship.

**Reply by chrisjlocke** [reply]
So it looks like its a semi-colon (;) separated file, not a comma separated, and numbers have a comma as a delimiter, so 0,00 rather than 0.00 - would this be correct?
If so, you might be able to import it OK within SQLite using those options - it can be tweaked.

**Reply by chrisjlocke** [reply]
![image](https://user-images.githubusercontent.com/7872218/122049978-ed232480-cdda-11eb-942b-c797ae303904.png)

**Reply by kkerklaan** [reply]
That is strange. When I exported it via access, I go to export, textfile, and then change txt to csv, can it be done in any other way?

**Reply by chrisjlocke** [reply]
This would also explain the '23/3/96' in the wrong column I mentioned...

**Reply by kkerklaan** [reply]
I see it now! thanks!

**Reply by chrisjlocke** [reply]
> and then change txt to csv

That doesn't change how the actual data is exported.
If you change the import settings, DB4S should accept that file OK.  👍

**Reply by kkerklaan** [reply]
One more question, how do I change field2 to putnr?

**Reply by chrisjlocke** [reply]
When you import, you can set a table name.  (see my screenshot above - first field) or you can go into 'database structure' and right click the table.

![image](https://user-images.githubusercontent.com/7872218/122053109-5b1d1b00-cdde-11eb-81b8-86ba29a7583e.png)

**Reply by chrisjlocke** [reply]
Access has many field types for data, while SQLite only has three main ones.

![image](https://user-images.githubusercontent.com/7872218/122053216-74be6280-cdde-11eb-9abf-f0ac878bb4c7.png)

Integer is any whole number - no decimal values.
Text is ... well, text. 
Blob is binary data - this can generally be ignored
Real is the one you mainly want - this is any number with decimals, eg, 1.234 (or 1,234 in your case)
Numeric can be sort of ignored too - this is mainly for compatibility with other databases.  Real should be 'good enough' for your use.

**Reply by chrisjlocke** [reply]
I'm sort of building an SQLite tutorial, so if you're scratching your head over anything, please yell and we/I can assist.  Happy to help out.

As another aside, I should add the obligatory thanks for using DB4S! Welcome to the community!

**Comment by DanGitR** [comment]
@kkerklaan I found this [explanation](https://www.pauldesalvo.com/easily-append-rows-to-a-table-from-multiple-csv-files-using-db-browser-for-sqlite/) helpful to add or append records to a table by  importing from a  .csv file.

**Comment by mgrojo** [comment, marked as answer]
It seems your Access is using a local convention for real numbers which uses a comma as decimal separator and, consequently, it is using semicolon for separating columns, instead of comma. If you configure DB Browser to use those separators while importing the csv file, you will get your data correctly. For the decimal separator, I don't remember right now if you can select the character or you set it to use the local convention and then it should work.

Edit: I hadn't read the hidden answers. This was already clarified, so maybe this was redundant.
